cpp/src/barretenberg/serialize/msgpack_apply.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include "msgpack.hpp"
+#include "msgpack_impl/drop_keys.hpp"
+
+namespace msgpack {
+/**
+ * @brief Passthrough method for better error reporting.
+ */
+template <msgpack_concepts::HasMsgPack T> void msgpack_apply(const auto& func, auto&... args)
+{
+    std::apply(func, msgpack::drop_keys(std::tie(args...)));
```

Comment:
Can you expand on this comment a bit (or the fn below)? What does it mean that they are passthrough methods and how might they be used?